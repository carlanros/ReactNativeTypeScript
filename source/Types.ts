import { NavigatorTypes } from './navigator/';
import { LoginTypes } from './login/';
import { UtilityTypes } from './utility/';
import { MediaTypes, TakePhoto } from './media';
import { StorageTypes, SaveByKey, RemoveKey, GetByKey } from './storage';

/**
 * The global application state as contained in Redux
 *
 * All state objects might not be available.
 *
 * @export
 * @interface appTypes.IApplicationState
 */
export interface IApplicationState {
  route?: NavigatorTypes.INavigationState;
  login?: LoginTypes.ILoginState;
  utility?: UtilityTypes.IUtilityState;
  storage?: StorageTypes.IStorageState;
  media?: MediaTypes.IMediaState;
}

/**
 * Describes possible props for a components.
 *
 * All actions and state objects might not be available.
 *
 * @export
 * @interface IProps
 */
export interface IProps extends IApplicationState {
  // AppLoadingChanged?: appActions.AppLoadingChanged;
  dispatch?: any;
  FacebookLogin?: LoginTypes.FacebookLogin;
  AppErrorChanged?: UtilityTypes.IAppErrorChanged;
  AppLoadingChanged?: UtilityTypes.IAppLoadingChanged;
  TakePhoto?: TakePhoto;
  GoogleLogin?: LoginTypes.GoogleLogin;
  GetByKey?: GetByKey;
  RemoveKey?: RemoveKey;
  SaveByKey?: SaveByKey;
}
