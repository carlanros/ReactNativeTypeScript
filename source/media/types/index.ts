// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
