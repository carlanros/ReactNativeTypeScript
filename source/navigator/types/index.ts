export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
