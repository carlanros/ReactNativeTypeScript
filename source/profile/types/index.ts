// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
