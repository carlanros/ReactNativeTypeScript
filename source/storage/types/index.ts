// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}