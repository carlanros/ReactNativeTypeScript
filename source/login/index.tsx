// login/index
import LoginContainer, { Login } from './components';
import LoginReducer from './reducers';
import * as LoginTypes from './types';
import { FacebookLogin } from './actions';

export { LoginContainer, Login, LoginReducer, LoginTypes, FacebookLogin };
