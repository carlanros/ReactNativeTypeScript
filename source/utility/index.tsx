// utility/index
import UtilityReducer from './reducers';
import * as UtilityTypes from './types';
import BusyIndicator, {
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
} from './components/BusyIndicator';
import { AppErrorChanged, AppLoadingChanged } from './actions';

export {
  AppErrorChanged,
  AppLoadingChanged,
  UtilityReducer,
  UtilityTypes,
  BusyIndicator,
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
};
