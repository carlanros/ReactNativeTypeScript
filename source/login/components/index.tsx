import React from 'react';
import { View, Text, TextInput } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect, Dispatch } from 'react-redux';

import * as types from '../../Types';
import * as Navigator from '../../navigator/components';
import Theme from '../../styles';

export class Login extends React.Component<types.IProps, {}> {
  render() {
    return (
      <Navigator.ModalBase hideClose>
        <View style={[Theme.alignCentered, Theme.paddingDefault]}>
          <Text style={[Theme.alignTop, Theme.paddingDefault]}>Login</Text>
          <TextInput
            style={[Theme.alignTop, Theme.paddingDefault, { width: 200 }]}
            placeholder='Username'
            autoFocus
            keyboardType='email-address'
          />
          <TextInput
            style={[Theme.alignTop, Theme.paddingDefault, { width: 200 }]}
            placeholder='Password'
            secureTextEntry
            onChangeText={name => this.onTextChanged(name)}
          />
        </View>
      </Navigator.ModalBase>
    );
  }

  onTextChanged = (text: string) => {
    this.userName = text;
    if (text && text.length > 5) {
      Actions.pop();
    }
  }

  userName: string = 'unknown';
}

const mapStateToProps = (state: types.IApplicationState) => ({
  route: state.route,
});

const mapDispatchToProps = (dispatch: Dispatch<types.IProps>) => ({
  dispatch,
});

export default connect<types.IApplicationState, types.IProps>(
  mapStateToProps,
  mapDispatchToProps
)(Login);
