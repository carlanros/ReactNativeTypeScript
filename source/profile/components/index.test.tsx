// profile/components/index.test
/// <reference types="jest"/>
import * as React from 'react';
import * as enzyme from 'enzyme';

import * as types from '../../Types';
import * as ProfileTypes from '../types';
import { LoginTypes } from '../../login';
import { ProfileBase } from '.';
import { BusyIndicator } from '../../utility';

jest.unmock('react-native');

// Navigator will call connected components
jest.mock('../../navigator', () => ({
  Routes: 'View',
  AppNavigator: 'Button',
}));

function setup() {
  const user: LoginTypes.IUser = {
    accessToken: '1',
    name: 'test',
    picture: '',
    type: 'Google',
  };
  const props: ProfileTypes.IProfileProps = {
    RemoveKey: jest.fn(),
    login: { isLoggedIn: false },
    utility: { isBusy: false, busyReason: null, hasError: false },
    storage: {
      key: '',
      value: user,
    },
  };

  // @ts-ignore
  const enzymeWrapper = enzyme.shallow(<ProfileBase {...props} />);
  const instance = enzymeWrapper.instance() as ProfileBase;

  return {
    props,
    enzymeWrapper,
    instance,
  };
}

describe('components', () => {
  describe('Profile', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();

      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
