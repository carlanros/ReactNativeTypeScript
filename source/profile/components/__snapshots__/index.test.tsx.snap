// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Profile should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileBase
    RemoveKey={[MockFunction]}
    login={
      Object {
        "isLoggedIn": false,
      }
    }
    storage={
      Object {
        "key": "",
        "value": Object {
          "accessToken": "1",
          "name": "test",
          "picture": "",
          "type": "Google",
        },
      }
    }
    utility={
      Object {
        "busyReason": null,
        "hasError": false,
        "isBusy": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ScrollView
        testID="scrollView"
      >
        <View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              test
            </Text>
            <UNDEFINED
              activeOpacity={0.7}
              large={true}
              rounded={true}
              source={
                Object {
                  "uri": "",
                }
              }
            />
          </View>
          <UNDEFINED
            icon={
              Object {
                "name": "sign-out",
                "type": "font-awesome",
              }
            }
            medium={true}
            onPress={undefined}
            style={
              Object {
                "margin": 10,
              }
            }
            title="Sign out"
          />
        </View>
      </ScrollView>,
      "style": Object {
        "display": "flex",
        "padding": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              test
            </Text>
            <UNDEFINED
              activeOpacity={0.7}
              large={true}
              rounded={true}
              source={
                Object {
                  "uri": "",
                }
              }
            />
          </View>
          <UNDEFINED
            icon={
              Object {
                "name": "sign-out",
                "type": "font-awesome",
              }
            }
            medium={true}
            onPress={undefined}
            style={
              Object {
                "margin": 10,
              }
            }
            title="Sign out"
          />
        </View>,
        "testID": "scrollView",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                test
              </Text>
              <UNDEFINED
                activeOpacity={0.7}
                large={true}
                rounded={true}
                source={
                  Object {
                    "uri": "",
                  }
                }
              />
            </View>,
            <UNDEFINED
              icon={
                Object {
                  "name": "sign-out",
                  "type": "font-awesome",
                }
              }
              medium={true}
              onPress={undefined}
              style={
                Object {
                  "margin": 10,
                }
              }
              title="Sign out"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  test
                </Text>,
                <UNDEFINED
                  activeOpacity={0.7}
                  large={true}
                  rounded={true}
                  source={
                    Object {
                      "uri": "",
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "test",
                  "ellipsizeMode": "tail",
                },
                "ref": null,
                "rendered": "test",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeOpacity": 0.7,
                  "large": true,
                  "rounded": true,
                  "source": Object {
                    "uri": "",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": undefined,
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": Object {
                "name": "sign-out",
                "type": "font-awesome",
              },
              "medium": true,
              "onPress": undefined,
              "style": Object {
                "margin": 10,
              },
              "title": "Sign out",
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ScrollView
          testID="scrollView"
        >
          <View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                test
              </Text>
              <UNDEFINED
                activeOpacity={0.7}
                large={true}
                rounded={true}
                source={
                  Object {
                    "uri": "",
                  }
                }
              />
            </View>
            <UNDEFINED
              icon={
                Object {
                  "name": "sign-out",
                  "type": "font-awesome",
                }
              }
              medium={true}
              onPress={undefined}
              style={
                Object {
                  "margin": 10,
                }
              }
              title="Sign out"
            />
          </View>
        </ScrollView>,
        "style": Object {
          "display": "flex",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                test
              </Text>
              <UNDEFINED
                activeOpacity={0.7}
                large={true}
                rounded={true}
                source={
                  Object {
                    "uri": "",
                  }
                }
              />
            </View>
            <UNDEFINED
              icon={
                Object {
                  "name": "sign-out",
                  "type": "font-awesome",
                }
              }
              medium={true}
              onPress={undefined}
              style={
                Object {
                  "margin": 10,
                }
              }
              title="Sign out"
            />
          </View>,
          "testID": "scrollView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  test
                </Text>
                <UNDEFINED
                  activeOpacity={0.7}
                  large={true}
                  rounded={true}
                  source={
                    Object {
                      "uri": "",
                    }
                  }
                />
              </View>,
              <UNDEFINED
                icon={
                  Object {
                    "name": "sign-out",
                    "type": "font-awesome",
                  }
                }
                medium={true}
                onPress={undefined}
                style={
                  Object {
                    "margin": 10,
                  }
                }
                title="Sign out"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    test
                  </Text>,
                  <UNDEFINED
                    activeOpacity={0.7}
                    large={true}
                    rounded={true}
                    source={
                      Object {
                        "uri": "",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "test",
                    "ellipsizeMode": "tail",
                  },
                  "ref": null,
                  "rendered": "test",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeOpacity": 0.7,
                    "large": true,
                    "rounded": true,
                    "source": Object {
                      "uri": "",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": undefined,
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "icon": Object {
                  "name": "sign-out",
                  "type": "font-awesome",
                },
                "medium": true,
                "onPress": undefined,
                "style": Object {
                  "margin": 10,
                },
                "title": "Sign out",
              },
              "ref": null,
              "rendered": null,
              "type": undefined,
            },
          ],
          "type": [Function],
        },
        "type": [MockFunction],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
